micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoCommandUtil.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2021 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.mongodb;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import io.micrometer.core.instrument.util.StringUtils;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+
+/**
+ * Provides utilities for MongoDB {@link BsonDocument commands}.
+ *
+ * @author Chris Bono
+ * @since 1.8.0
+ */
+class MongoCommandUtil {
+
+    // See https://docs.mongodb.com/manual/reference/command for the command reference
+    private static final Set<String> COMMANDS_WITH_COLLECTION_NAME = new HashSet<>(Arrays.asList(
+            "aggregate", "count", "distinct", "mapReduce", "geoSearch", "delete", "find", "findAndModify",
+            "insert", "update", "collMod", "compact", "convertToCapped", "create", "createIndexes", "drop",
+            "dropIndexes", "killCursors", "listIndexes", "reIndex"));
+
+    /**
+     * Attempts to determine the name of the collection a command is operating on.
+     *
+     * <p>Because some commands either do not have collection info or it is problematic to determine the collection info,
+     * there is an allow list of command names {@link #COMMANDS_WITH_COLLECTION_NAME} used. If {@code commandName} is
+     * not in the allow list or there is no collection info in {@code command}, it will use the content of the
+     * {@code 'collection'} field on {@code command}, if it exists.
+     *
+     * <p>Taken from <a href="https://github.com/openzipkin/brave/blob/master/instrumentation/mongodb/src/main/java/brave/mongodb/TraceMongoCommandListener.java#L115>TraceMongoCommandListener.java</a>
+     *
+     * @param commandName name of the mongo command
+     * @param command mongo command object
+     * @return optional collection name or empty if could not be determined or not in the allow list of command names
+     */
+    Optional<String> determineCollectionName(String commandName, BsonDocument command) {
+        if (COMMANDS_WITH_COLLECTION_NAME.contains(commandName)) {
+            Optional<String> collectionName = getNonEmptyBsonString(command.get(commandName));
+            if (collectionName.isPresent()) {
+                return collectionName;
+            }
```

Comment:
Please check my comment here: https://github.com/micrometer-metrics/micrometer/pull/2337#discussion_r660961391

Reply:
Do you mind resolving this one now?